### ISSUE TITLE
Add clustering to member map

### ISSUE THREAD

**Issue by danieltott** [Virtual-Coffee/virtualcoffee.io#1031]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue Context

On the new [members map](https://virtualcoffee.io/members), it would be cool to get some clustering going instead of individual markers. That would make it easier to scan areas.

Also, we're running in to an issue where if two markers are at the same location, only the top one is visible (zoom in to the Los Angelos area, both @jdwilkin4 and @brennakh are in LA, but only Jessica's marker is visible).

Hopefully the clustering thing would solve that as well.

### Proposed solution

[`react-leaflet-cluster`](https://github.com/akursat/react-leaflet-cluster) seems like a really good bet here, but if there's a better option I'm all ears

### Alternatives Considered

¯\_(ツ)_/¯

### Additional Resources

- We're using [`react-leaflet`](https://react-leaflet.js.org/) to drive the map
- `react-leftleft` is a React wrapper around [`Leaflet`](https://leafletjs.com/)
- The code that drives the map can be found here: https://github.com/Virtual-Coffee/virtualcoffee.io/blob/main/app/components/MemberMap.tsx

### Code of Conduct

- [X] I've read the Code of Conduct and understand my responsibilities as a member of the Virtual Coffee community

**Comment by ARodriguezHacks** [comment]
Can I give this a try?

**Comment by danieltott** [comment]
@ARodriguezHacks It's all you!

**Comment by ARodriguezHacks** [comment]
@danieltott this issue is 95% done except for a few things:

1. Asking the maintainers to provide design input for individual contributor icon and the cluster icon. The way the 'react-leaflet-cluster' works, from what it seems, the default icon for the `Marker` component from 'react-leaflet' doesn't work as expected. Unless I'm doing something wrong. So in other words, my branch is using temporary custom icons, one borrowed from the 'react-leaflet-cluster' code example and another from the VC assets folder (this one being used for the individual contributor).
2. There's an type error with the `createClusterCustomIcon` function borrowed from 'react-leaflet-cluster' code example. The example shows code importing `MarkerCluster` from 'leaflet', but that was for version ^1.8.0; VC is using version ^1.9.4, but I can't seem to import `MarkerCluster` since it gives an error that it's not included in the 'leaflet' library. 

![react_leaflet_cluster_code_sample](https://github.com/Virtual-Coffee/virtualcoffee.io/assets/22532097/f981bd7e-d1e9-4b9e-ada2-2f91640225ca)
![type_import_error](https://github.com/Virtual-Coffee/virtualcoffee.io/assets/22532097/67d93041-c493-4aa0-ba1f-77c26a996b77)

**Comment by ARodriguezHacks** [comment]
Also here's a video for the latest functionality with the current/temporary icons and styling that I think need design input from the VC maintainers:

https://github.com/Virtual-Coffee/virtualcoffee.io/assets/22532097/189165d3-04e4-4731-b64b-b243f5530d15

**Comment by pedaars** [comment]
It looks good @ARodriguezHacks I have been playing around with this as it seemed interesting but I couldn't even get the map to load or if I did it just had the numbers visible with no icons.

**Comment by danieltott** [comment]
@ARodriguezHacks I think this is great to start!

For `MarkerCluster`, try this:

```ts
import type L from 'leaflet';

const createClusterCustomIcon = function (cluster: L.MarkerCluster) {
  //...
}
```

Assuming that works, I'd say go ahead and create a PR so we can see it in action!

**Comment by danieltott** [comment]
closed via #1109
